### PR TITLE
disabled emoji highlighting to fix bug

### PR DESCRIPTION
**Pull request by rahil1304**
<!--

Please prefix the title with the platform you are targetting:

Here are some examples of good titles:

- Desktop: Fixes#3796 : Fixed emoji highlighting bug

### CODE REVIEW

**Comment by CalebJohn** [comment]
I see that there has been some misunderstanding here. I misunderstood what you asked. This is not the correct fix, but it is close.

If you check under settings you will see the that there is a plugin options screen
image

The value you set to false shouldn't be false, it should follow the setting in this toggle. So that the use can enable/disable the highlighting.

**Comment by rahil1304** [comment]
okay got it. My bad. Will make the required changes @CalebJohn

**Comment by rahil1304** [comment]
Sorry, I'm a little new to this so yet a little confused. So essentially I create a new option to enable/disable the highlighting. 
So the user can check the checkbox if they want the highlighting and uncheck it if they don't want it. Correct?
Also there a plugin files in different languages. So if I create a new plugin option, would I need to create a new option for each language?

**Comment by CalebJohn** [comment]
No problem, I appreciate your effort.
You actually don't need to create a new setting, which will save you some effort. As you can see the setting already exists
image

But it's just not properly connected to the codemirror editor. What you will need to do is find out what the setting name is for the emoji plugin. Once you have the name you can use the function `Setting.value` to read that value and disable/enable the emoji support in codemirror.
An example of what that might look like is below.

```diff
-emoji: false,
+emoji: Setting.value('The name of the plugin emoji would go here'),
```

As a hint to what the settings name is, you can try checking line 480 of `ReactNativeClient/lib/models/Setting.js` :)

**Comment by rahil1304** [comment]
Hey, thank you so much for the help @CalebJohn. I really appreciate it. 
I have made the required changes to this same branch. Do let me know if anything else is required.

File: ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useJoplinMode.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import 'codemirror/addon/mode/multiplex';
 import 'codemirror/mode/stex/stex';
+const Setting = require('../../../../../lib/models/Setting.js');
```
Comment (CalebJohn) [inline ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useJoplinMode.ts:3]: Almost there! 

However, you don't need to require Settings using the relative levels. For example check `ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx` for an example of how `Setting` should be required.

Comment (rahil1304) [reply ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useJoplinMode.ts:3]: So I checked that. But the lib folder is in a different folder directory. So shouldn't the ideal way be to use `const Setting = require('../../../../../lib/models/Setting.js');` and not `const Setting = require('lib/models/Setting.js');` @CalebJohn

Comment (rahil1304) [reply ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useJoplinMode.ts]: Done. Corrected it @CalebJohn

**Comment by CalebJohn** [comment]
LGTM, @laurent22 do you mind taking a look at this?

**Comment by laurent22** [comment]
Thanks for looking into it @rahil1304 and @CalebJohn, I think your solution makes sense. Let's wait till version 1.3 to merge because the Setting class is now TypeScript so would need to be imported using `import`.

**Comment by laurent22** [comment]
Ok, v1.3 pre-release is coming soon so let's merge.

**Pull request merged**